cmd/cluster/core/create_test.go

```diff
@@ -0,0 +1,89 @@
+package core
+
+import (
+	"context"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	apiversion "k8s.io/apimachinery/pkg/version"
+	fakediscovery "k8s.io/client-go/discovery/fake"
+	fakekubeclient "k8s.io/client-go/kubernetes/fake"
+)
+
+func TestValidateMgmtClusterAndNodePoolCPUArchitectures(t *testing.T) {
+	ctx := context.Background()
+
+	fakeKubeClient := fakekubeclient.NewSimpleClientset()
+	fakeDiscovery, ok := fakeKubeClient.Discovery().(*fakediscovery.FakeDiscovery)
+
+	if !ok {
+		t.Fatalf("failed to convert FakeDiscovery")
+	}
+
+	// if you want to fake a specific version
+	fakeDiscovery.FakedServerVersion = &apiversion.Info{
+		Platform: "linux/amd64",
+	}
+
+	tests := []struct {
+		name        string
+		opts        *RawCreateOptions
+		expected    bool
+		expectError bool
+	}{
+		{
+			name: "Test case 1: Valid multi-arch image",
```

Comment:
Nice!
nit: no need to prefix with "test case n:"
Also can we rephrase to follow convention "When x it should y"? e.g. When a multi arch release image is given it should succeed with any NodePool Arch